modules/fundamental/src/vulnerability/model/details/vulnerability_advisory.rs

```diff
@@ -431,56 +439,246 @@ impl VulnerabilityAdvisoryStatus {
+                        entity::sbom::Column::Sha256,
+                        entity::sbom::Column::Sha384,
+                        entity::sbom::Column::Sha512,
+                        entity::sbom::Column::Labels,
+                        entity::sbom::Column::DocumentId,
+                        entity::sbom::Column::Published,
+                        entity::sbom::Column::Authors,
+                    ])
+                    .column_as(entity::cpe::Column::Id, "cpe_id")
+                    .columns([
+                        entity::cpe::Column::Part,
+                        entity::cpe::Column::Vendor,
+                        entity::cpe::Column::Product,
+                        entity::cpe::Column::Version,
+                        entity::cpe::Column::Update,
+                        entity::cpe::Column::Edition,
+                        entity::cpe::Column::Language,
+                    ])
+                    .into_model::<SbomDetailsCatcher>()
+                    .one(tx)
+                    .await?
+            {
+                let sbom_cpe = sbom_details.cpe_id.and_then(|_| {
+                    let mut builder = cpe::uri::Uri::builder();
+                    if let Some(part) = sbom_details.part.as_ref() {
+                        if part != "*" {
+                            builder.part(part);
+                        }
+                    }
+                    if let Some(vendor) = sbom_details.vendor.as_ref() {
+                        if vendor != "*" {
+                            builder.vendor(vendor);
+                        }
+                    }
+                    if let Some(product) = sbom_details.product.as_ref() {
+                        if product != "*" {
+                            builder.product(product);
+                        }
+                    }
+                    if let Some(version) = sbom_details.version.as_ref() {
+                        if version != "*" {
+                            builder.version(version);
+                        }
+                    }
+                    if let Some(update) = sbom_details.update.as_ref() {
+                        if update != "*" {
+                            builder.update(update);
+                        }
+                    }
+                    if let Some(edition) = sbom_details.edition.as_ref() {
+                        if edition != "*" {
+                            builder.edition(edition);
+                        }
+                    }
+                    if let Some(language) = sbom_details.language.as_ref() {
+                        if language != "*" {
+                            builder.language(language);
+                        }
+                    }
+                    if let Ok(cpe) = builder.validate() {
+                        Some(cpe.to_owned())
+                    } else {
+                        None
+                    }
+                });
```

Comment:
This feels verbose. Could things be simpler with a `cpe()` method on `sbom_details`?

Reply:
IIRC we already have that already, including tests and all workarounds: https://github.com/trustification/trustify/blob/cf8372a94ea5742967ac8e6262248509e2ed6f18/entity/src/cpe.rs#L152-L174

Reply:
Maybe it makes sense to turn this into a trait (IntoCpeDetails`)? Which only returns CPE information (like part, vendor, …) and then implement `TryInto<Cpe>` (or `TryFrom<T> for Cpe` for that trait. Having all the fixes and workarounds in a single place.